File: java/162.FindPeakElement.java
public class Solution {
	public int findPeakElement(int[] nums) {
		if (nums.length == 1) return 0;
		if (nums[0]>nums[1]) return 0;
		int l = nums.length;
		if (nums[l-1] > nums[l-2]) return l-1;
		int left = 0;
		int right = l-1;
		int mid;
		while (left<right){
			mid = (left+right)/2;
			if (nums[mid-1]<nums[mid] && nums[mid]>nums[mid+1]) return mid;
			if (nums[mid-1]<nums[mid] && nums[mid]<nums[mid+1]) left = mid;
			else right = mid;
		}
		return right;
	}
}

File: python/162.FindPeakElement.py
class Solution(object):
	def findPeakElement(self, nums):
		"""
		:type nums: List[int]
		:rtype: int
		"""
		if len(nums)==1: return 0
		if nums[0] > nums[1]: return 0
		if nums[-1] > nums[-2]: return len(nums)-1
		left = 0
		right = len(nums) - 1
		while left<right:
			mid = (left+right)//2
			if nums[mid-1]<nums[mid] and nums[mid]>nums[mid+1]: return mid
			if nums[mid-1]<nums[mid] and nums[mid]<nums[mid+1]: left = mid
			else: right = mid
		return right